Add App Translation Strings in Settings

azrikahar opened the pull request:
## Changes

- Allow adding custom translation strings in settings for app-wide translations
- Adds `input-traslated-strings` input interface for choosing custom translation strings

## Preview

https://user-images.githubusercontent.com/42867097/158379099-6f29583c-2046-48b2-826f-c748939c8cd7.mp4

---

Closes directus/organization#76

azrikahar committed:
add migration for translation strings

azrikahar committed:
add to navigation

azrikahar committed:
WIP

azrikahar committed:
fix dialog overflow

azrikahar committed:
update translation keys

azrikahar committed:
Update logic

azrikahar committed:
add placeholder to system-language

azrikahar committed:
fix translation

azrikahar committed:
remove unused import

azrikahar committed:
reset dialog on create new

azrikahar committed:
ensure search input is visible when searching

azrikahar committed:
merge translation strings on set language

azrikahar committed:
merge translation strings on update

azrikahar committed:
hydrate

azrikahar committed:
Merge branch 'main' into feat/translation-strings

danieljwestman commented:
This will be a huge improvement for us with multi lang. Great work! Seems ready to be merged? 🤓

azrikahar committed:
make sure null translation do not get merged

azrikahar committed:
change dialog to drawer

azrikahar committed:
update placeholder text

azrikahar committed:
fix form value

azrikahar committed:
revert dialog style change

azrikahar committed:
rename drawer component

rijkvanzanten committed:
Merge branch 'main' into feat/translation-strings

rijkvanzanten commented on app/src/hydrate.ts:
```diff
@@ -16,6 +16,7 @@ import {
 	useUserStore,
 	useNotificationsStore,
 } from '@/stores';
+import { useTranslationStrings } from '@/modules/settings/composables/use-translation-strings';
```
Seeing that this composable is used outside of the module, lets move this composable to the root composables folders

rijkvanzanten commented on app/src/hydrate.ts:
```diff
@@ -50,6 +51,7 @@ export async function hydrate(): Promise<void> {
 	const appStore = useAppStore();
 	const userStore = useUserStore();
 	const permissionsStore = usePermissionsStore();
+	const { refresh: hydrateTranslationStrings } = useTranslationStrings();
```
Maybe we can just rename the exported value from `useTranslationStrings`. Makes it easier to find and replace later if needed

rijkvanzanten replied:
Nah, forget it. It's cleaner to leave it as refresh in the context of that composable 👍🏻

rijkvanzanten committed:
Force safe key name

rijkvanzanten committed:
Move interface to system interfaces

The saved values are Directus app proprietary, so to prevent confusion in what it's supposed to do, we'll move it to system.

rijkvanzanten committed:
Move composable to root composables

rijkvanzanten committed:
Use new languages input in interface/display options

rijkvanzanten reviewed (approved):
LGTM!

@azrikahar Could you do another pass on the display/interface options to see if I missed any options that should use the new interface (and maybe see if we can update `list` to also use it?) Feel free to merge this afterwards 🙂

azrikahar committed:
hide translation strings field in project settings

azrikahar committed:
set system true to system-input-translated-string

azrikahar committed:
use this in field detail notes

azrikahar commented:
@rijkvanzanten here are the following changes, will leave the merging to you once you have reviewed them 👍 (feel free to ignore some/all of the changes that are correct)

- Noticed the `Translation Strings` field is showing up at the end of the Project Settings page, opted to hide it here. However I can also see the value of being able to copy/paste the raw value, but then again maybe this is another potential enhancement that should (somehow) be doable in the Translation Strings page instead.

  Preview of Translation Strings if not hidden:
  
  ![chrome_YV1oPqmM3o](https://user-images.githubusercontent.com/42867097/159471260-789b2daf-b57b-4bed-9744-fac95d2aa69e.png)
  
- Noticed the interface is showing up in string fields, but since it was moved to system interfaces as you mentioned, added `system: true` to hide it. 

- Used this new interface for field detail notes:

  ![chrome_bq6f8Is4dk](https://user-images.githubusercontent.com/42867097/159472049-c74b42b1-fb17-4343-ada1-7733a800fa35.png)
 
- For the list or repeater, added the new interface for "Create New" label and "note" of each field:

  "Create New" label:
  
   ![chrome_Fs8Fa0uY8N](https://user-images.githubusercontent.com/42867097/159472227-875c3237-6d59-4e01-b59d-75afbd3a39f7.png)

  Notes of each field:
  
   ![chrome_kYMJP578QC](https://user-images.githubusercontent.com/42867097/159472291-8b3fffe8-96b9-491c-92af-c36db8896b9a.png)

azrikahar committed:
use in list options
